src/main/scala/org/apache/mesos/chronos/scheduler/api/JobManagementResource.scala

```diff
@@ -201,13 +202,19 @@ class JobManagementResource @Inject()(val jobScheduler: JobScheduler,
   @Timed
   def list(): Response = {
     try {
-      val jobs = ListBuffer[BaseJob]()
       import scala.collection.JavaConversions._
-      jobGraph.dag.vertexSet().map({
-        job =>
-          jobs += jobGraph.getJobForName(job).get
-      })
-      Response.ok(jobs.toList).build
+      val jobs = jobGraph.dag.vertexSet()
+        .map { jobGraph.getJobForName }
+        .flatten
+        .map { // copies fetch in uris or uris in fetch (only one can be set) __only__ in REST get, for compatibility
+          case j : ScheduleBasedJob  =>
+            if(j.fetch.isEmpty) j.copy(fetch = j.uris.map { Fetch(_) })
+            else                j.copy(uris =  j.fetch.map { _.uri })
+          case j : DependencyBasedJob =>
+            if(j.fetch.isEmpty) j.copy(fetch = j.uris.map { Fetch(_) })
```

Comment:
This cannot be moved to a method, due to the fact that there is a case class duplicate between `ScheduleBasedJob` and `DependencyBasedJob`. An alternative would be to add one case class standard copy() in the trait, but that would be act just to remove this warning.

Best course of action would be to refactor to remove duplicates in the model (out of scope of the present PR).